main.go

```diff
@@ -149,6 +157,42 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Subscription controller - skipping subscription controller if preflights are bypassed.
+	if !helper.IsPreflightBypassed() {
+		restConfig := ctrl.GetConfigOrDie()
+		restConfig.Timeout = time.Second * 10
+		k8sclient, err := client.New(restConfig, client.Options{
+			Scheme: mgr.GetScheme(),
+		})
+		if err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "Subscription")
+			os.Exit(1)
+		}
+		// For subscription controller, the watchnamespace is always the operator namespace
+		operatorInstallationNamespace, err := helper.GetOperatorNamespace()
+		if err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "Subscription")
+			os.Exit(1)
+		}
+		setupLog.Info(fmt.Sprintf("Operator Namespace is: %s", operatorInstallationNamespace))
+		discoveryClientSubscription, err := discovery.NewDiscoveryClientForConfig(mgr.GetConfig())
+		if err != nil {
+			setupLog.Error(err, "unable to create discovery client")
+			os.Exit(1)
+		}
+		if err = (&subcontroller.SubscriptionReconciler{
```

Comment:
@carlkyrillos any ideas if there's anything available in your PR (bump the deps) to help us with this issue?

Reply:
@MStokluska I haven't tested it myself but controller-runtime [v0.16.0](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.16.0) introduced more [granular controls](https://github.com/kubernetes-sigs/controller-runtime/pull/2421) over the cache which should allow you to watch subscriptions in all namespaces while being otherwise namespace-scoped for everything else. Here is where the cache is configured in my [PR](https://github.com/3scale/3scale-operator/pull/936/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R115-R119)

Reply:
That's solved with the new manager options. Thanks!